dapp/src/components/name-record/AvatarDisplay.tsx

```diff
@@ -1,45 +1,61 @@
+
+import { nftDisplayVariants } from './variants';
 
-interface AvatarDisplayProps {
-    registration: RegistrationNft;
+interface AvatarDisplayProps extends NftDisplayProps {
+    button?: React.ReactNode;
 }
 
-export function AvatarDisplay({ registration }: AvatarDisplayProps) {
-    const {
-        data,
-        isLoading: isLoadingRegistration,
-        isError: isErrorRegistration,
-    } = useNameRecord(registration.name);
+export function AvatarDisplay({ registrationNft, size, badge, button }: AvatarDisplayProps) {
+    const [showAvatar, setShowAvatar] = useState(false);
+
+    const { data } = useNameRecord(registrationNft.name);
+
+    //TODO: Remove when we add the svg api
+    const PLACEHOLDER_DISPLAY = `/placeholder-name-display.svg`;
 
     const avatarId = data?.type === 'unavailable' ? data?.nameRecord.avatar : null;
 
-    const {
-        data: avatarObject,
-        isLoading: isLoadingAvatarObject,
-        isError: isErrorAvatarObject,
-    } = useGetObject({ id: avatarId ?? '', options: { showDisplay: true, showContent: true } });
-
-    const mediaUrl =
-        isLoadingAvatarObject || isLoadingRegistration || isErrorAvatarObject || isErrorRegistration
-            ? registration.imageUrl
-            : avatarObject?.display?.data?.image_url || registration.imageUrl;
-
-    return mediaUrl && avatarObject ? (
-        <img
-            src={mediaUrl}
-            alt={registration.name}
-            className="w-full h-full object-cover rounded-lg"
-        />
-    ) : (
-        <div className="w-full h-full bg-neutral-30/20 rounded-lg flex items-end justify-end">
-            <small className="text-neutral-50 m-xs">
-                {registration.name} - No avatar available
-            </small>
+    const { data: avatarObject } = useGetObject({
+        id: avatarId ?? '',
+        options: { showDisplay: true, showContent: true },
+    });
+
+    const avatarSrc = avatarObject?.display?.data?.image_url;
+
+    useEffect(() => {
+        if (!avatarSrc) return;
+
+        const img = new Image();
+        img.src = avatarSrc;
+
+        img.onload = () => setShowAvatar(true);
+        img.onerror = () => setShowAvatar(false);
+    }, [avatarSrc]);
+
+    return (
+        <div className={nftDisplayVariants({ size })}>
+            <div className="w-full h-full flex flex-col relative rounded-xl overflow-hidden">
+                {badge && <div className="absolute top-sm left-sm">{badge}</div>}
+                {button && (
+                    <div className="opacity-0 group-hover/display:opacity-100 transition-opacity absolute w-full top-0 right-0 px-sm py-sm bg-gradient-to-b from-black/80 to-transparent flex justify-end pointer-events-none">
+                        <div className="shadow-xl pointer-events-auto">{button}</div>
+                    </div>
+                )}
+
+                <img
+                    className="absolute inset-0 w-full h-full -z-[1]"
```

Comment:
the images look a little off :see_no_evil: 

![image](https://github.com/user-attachments/assets/f6e63cdc-de5e-4d93-8f83-614ad9f485e3)

```suggestion
                    className="absolute inset-0 w-full h-full -z-[1] object-cover"
```